Loop TodayExtension/TodayExtensionContext.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  TodayExtensionContext.swift
+//  Loop
+//
+//  Created by Bharat Mediratta on 11/25/16.
+//  Copyright © 2016 LoopKit Authors. All rights reserved.
+//
+//  This class allows Loop to pass context data to the Today Extension.
+
+import Foundation
+import HealthKit
+import LoopKit
+import InsulinKit
+import MinimedKit
+
+struct ReservoirValueContext: ReservoirValue {
+    var startDate: Date
+    var unitVolume: Double
+    var capacity: Int
+}
+
+struct LoopContext {
+    var dosingEnabled: Bool
+    var lastCompleted: Date?
+}
+
+struct NetBasalContext {
+    var rate: Double
+    var percentage: Double
+    var startDate: Date
+}
+
+struct SensorDisplayableContext: SensorDisplayable {
+    var isStateValid: Bool
+    var stateDescription: String
+    var trendType: GlucoseTrend?
+    var isLocal: Bool
+}
+
+struct GlucoseContext {
+    var latest: GlucoseValue
+    var sensor: SensorDisplayable?
+}
+
+class TodayExtensionContext {
```

Comment:
This class is trying to do more than one thing. On one hand you've got convenience methods on NSUserDefaults which are better in an `extension UserDefaults`. On the other hand you've got raw data serialization which belongs in struct that implements `RawRepresentable`.